app/controllers/concerns/hal_actions.rb

```diff
@@ -16,14 +16,14 @@ def show
   end
 
   def create
-    self.resource = consume!(resource)
+    consume_with_content_type! resource
```

Comment:
I'm surprised this works without the `self.resource` assignment, I remember needing that since subsequent calls assume that has been set here.